## Topic: Store the Local Data into MongoDB Atlas(cloud)

### OUTCOMES

- 1. Load the Dataset into local enviroment

- 2. Some Basic Operation

- 3. Load the Dataset into MongoDB Atlas (cloud)


### 1. Load the Dataset into local enviroment

In [1]:
print("ok")

ok


In [3]:
# import necessary libraries
import pandas as pd

In [4]:
# show the Present working directory
%pwd

'e:\\Data Science\\ALL Projects\\ML ALL Projects\\end-to-end-us-visa-approval-prediction\\notebooks'

- Key NOTE:
    - First we go to the root directory 
    
    - then access the Shipment-data.csv, 
        - command: ../data/Shipment-data.csv
            - where, 
                - ../ means -> go the Root Directory

In [5]:
# load the dataset
data = pd.read_csv("../data/Visadataset.csv")

data.head()

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified


### 2. Some Basic Operation

In [6]:
# Show first 5 rows
data.head()

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified


In [7]:
# show last 5 rows
data.tail()

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
25475,EZYV25476,Asia,Bachelor's,Y,Y,2601,2008,South,77092.57,Year,Y,Certified
25476,EZYV25477,Asia,High School,Y,N,3274,2006,Northeast,279174.79,Year,Y,Certified
25477,EZYV25478,Asia,Master's,Y,N,1121,1910,South,146298.85,Year,N,Certified
25478,EZYV25479,Asia,Master's,Y,Y,1918,1887,West,86154.77,Year,Y,Certified
25479,EZYV25480,Asia,Bachelor's,Y,N,3195,1960,Midwest,70876.91,Year,Y,Certified


In [8]:
# shape of the data
data.shape

(25480, 12)

- Short Note:
    - Total number of rows = 25480
    - Total number of columns = 12

In [9]:
# show the dtype and memory usage
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25480 entries, 0 to 25479
Data columns (total 12 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   case_id                25480 non-null  object 
 1   continent              25480 non-null  object 
 2   education_of_employee  25480 non-null  object 
 3   has_job_experience     25480 non-null  object 
 4   requires_job_training  25480 non-null  object 
 5   no_of_employees        25480 non-null  int64  
 6   yr_of_estab            25480 non-null  int64  
 7   region_of_employment   25480 non-null  object 
 8   prevailing_wage        25480 non-null  float64
 9   unit_of_wage           25480 non-null  object 
 10  full_time_position     25480 non-null  object 
 11  case_status            25480 non-null  object 
dtypes: float64(1), int64(2), object(9)
memory usage: 2.3+ MB


In [10]:
# type of dataset
type(data)

pandas.core.frame.DataFrame

### Convert Pandas DataFrame to Dictionary

- Because MongoDB store the data into json or dictionary format

- orient = "records": Give the well structure

In [10]:
data_dic = data.to_dict(orient = "records")

In [22]:
# length
len(data_dic)

25480

In [23]:
# Check type of data
type(data_dic)

list

### 3. Load the Dataset into MongoDB Atlas (cloud)

In [17]:
# import library
import pymongo

In [ ]:
# Database name
DB_NAME = "US_VISA_DB"
COLLECTION_NAME = "visa_applications"
CONNECTION_URL = "your connection url"


- short note:

    - DB_NAME = Database name

    - COLLECTION_NAME = Table name of the Database

    - CONNECTION_URL = Connection string between MongoDB Atlas and python

### Storing Data to MongoDB Database

In [25]:
client = pymongo.MongoClient(CONNECTION_URL)
data_base = client[DB_NAME]
collection = data_base[COLLECTION_NAME]

In [26]:
rec = collection.insert_many(data_dic)

- Key NOTE:
    - Upload our local Data to MangoDB atlas

### Fetching Data from MongoDB Database

In [27]:
# Fetching all records
records = collection.find()

In [28]:
# it give us an object of the data and it can be iterable
records

In [ ]:
# one one to display
# for i in records:
#     print(i)


### Convert records into DataFrame

In [29]:
# fetching all record from the MongoDB atlas and convert into dataframe
df = pd.DataFrame(list(collection.find()))

In [30]:
# show first 5 rows
df.head()

,_id,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,6a6f916358ee79f667bf5e81,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,6a6f916358ee79f667bf5e82,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,6a6f916358ee79f667bf5e83,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,6a6f916358ee79f667bf5e84,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,6a6f916358ee79f667bf5e85,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified


In [31]:
# Drop the _id columns
if "_id" in df.columns:
    df.drop(columns="_id", inplace=True)

- Key Note:
    - Because, MongoDB Create a unique Id for a dataset.

In [32]:
df.head()

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified


#### Convert DataFrame to CSV format

In [34]:
# convert DataFrame -> CSV
df.to_csv("visa_dataset.csv", index = False)

### Deleted the DataBase

In [ ]:
# Deleted Database
# client.drop_database("KzRaihan")